internal/helper/request_url.go

```diff
@@ -0,0 +1,21 @@
+package helper
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ConstructRequestURL constructs a request URL based on paths and query parameters.
+func ConstructRequestURL(base string, queryParameters url.Values, paths ...string) (url string) {
```

Comment:
can we change the function definition to something like, 
ConstructRequestURL(paths []string, queryParam url.Values) 

or

type requestURL string 
constructReqUrl(path ...string) requestURL
func (requestURL) AppendQueryParam (url.values) requestURL
func (requestURL) String () string


The idea behind suggesting this change is to avoid passing nil in the function parameters

Reply:
Sure